backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/work/impl/AnalyzeWork.java

```diff
@@ -0,0 +1,149 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.work.impl;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.hibernate.search.backend.elasticsearch.client.impl.Paths;
+import org.hibernate.search.backend.elasticsearch.client.spi.ElasticsearchRequest;
+import org.hibernate.search.backend.elasticsearch.client.spi.ElasticsearchResponse;
+import org.hibernate.search.backend.elasticsearch.gson.impl.JsonAccessor;
+import org.hibernate.search.backend.elasticsearch.gson.impl.JsonArrayAccessor;
+import org.hibernate.search.backend.elasticsearch.util.spi.URLEncodedString;
+import org.hibernate.search.engine.backend.analysis.AnalysisToken;
+import org.hibernate.search.util.common.AssertionFailure;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+public class AnalyzeWork extends AbstractNonBulkableWork<List<? extends AnalysisToken>> {
+	private static final JsonArrayAccessor TOKENS_ACCESSOR = JsonAccessor.root().property( "tokens" ).asArray();
+	private static final JsonAccessor<String> TOKEN_ACCESSOR = JsonAccessor.root().property( "token" ).asString();
+	private static final JsonAccessor<Integer> START_OFFSET_ACCESSOR =
+			JsonAccessor.root().property( "start_offset" ).asInteger();
+	private static final JsonAccessor<Integer> END_OFFSET_ACCESSOR = JsonAccessor.root().property( "end_offset" ).asInteger();
+	private static final JsonAccessor<String> TYPE_ACCESSOR = JsonAccessor.root().property( "type" ).asString();
+
+	private static final ElasticsearchRequestSuccessAssessor SUCCESS_ASSESSOR =
+			ElasticsearchRequestSuccessAssessor.builder().build();
+
+	private AnalyzeWork(Builder builder) {
+		super( builder );
+	}
+
+	@Override
+	protected List<? extends AnalysisToken> generateResult(ElasticsearchWorkExecutionContext context,
+			ElasticsearchResponse response) {
+		JsonObject body = response.body();
+
+		// we don't use Gson to keep ElasticsearchAnalysisToken immutable.
+		List<ElasticsearchAnalysisToken> tokens = new ArrayList<>();
+		for ( JsonElement element : TOKENS_ACCESSOR.getOrCreate( body, JsonArray::new ) ) {
+			JsonObject token = element.getAsJsonObject();
+			tokens.add( new ElasticsearchAnalysisToken(
+					TOKEN_ACCESSOR.get( token ).orElse( "" ),
```

Comment:
Is this even possible? If not I'd use `orElseThrow` here, and if yes... I don't know, I'd like to understand why.